test/updater_unittest.cc

```diff
@@ -0,0 +1,84 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+#include <string>
+#include <vector>
+
+namespace google {
+
+class UpdaterTest : public ::testing::Test {
+ protected:
+  // query_metadata function not needed to test callbacks.
+  UpdaterTest() : config(), store(config) {}
+
+  static bool ValidateConfiguration(MetadataUpdater* updater) {
+    return updater->ValidateConfiguration();
+  }
+
+  static void UpdateMetadataCallback(
+      MetadataUpdater* updater,
+      MetadataUpdater::ResourceMetadata&& result) {
+    updater->UpdateMetadataCallback(std::move(result));
+  }
+
+  static void UpdateResourceCallback(
+      MetadataUpdater* updater,
+      const MetadataUpdater::ResourceMetadata& result) {
+    updater->UpdateResourceCallback(result);
+  }
+
+  Configuration config;
+  MetadataStore store;
+};
+
+namespace {
+
+TEST_F(UpdaterTest, ValidateConfiguration) {
```

Comment:
Let's give this a better name, e.g., `OneMinutePollingIntervalIsValid`?
Can you also add a `ZeroPollingIntervalIsValid` test?

Reply:
Done.